src/Stratis.Bitcoin/P2P/Protocol/Payloads/ProvenHeadersPayload.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using NBitcoin.Protocol;
+
+namespace Stratis.Bitcoin.P2P.Protocol.Payloads
+{
+    /// <summary>
+    /// Proven block headers received after a getheaders message.
+    /// </summary>
+    [Payload("provhdr")]
+    public class ProvenHeadersPayload : Payload
+    {
+        private class ProvenBlockHeaderWithTxCount : IBitcoinSerializable
```

Comment:
Do we need this?

Reply:
`HeadersPayload` seems to have it. I guess it is used to track tx count

Reply:
Actually I don't see any meaningful usage of it. I removed it.

Reply:
Exactly